torch/distributed/rpc/api.py

```diff
@@ -238,9 +241,31 @@ def _all_gather(obj, timeout=UNSET_RPC_TIMEOUT):
                 f"after {timeout:.2f} seconds. The first exception is {errors[0][1]}"
             )
 
+    # Clean up for the states using the sequence_id
+    with _all_gather_dict_lock:
+        states = _all_gather_sequence_id_to_states.pop(sequence_id)
     return states.gathered_objects
 
 
+@_require_initialized
+def _barrier(worker_names):
+    r"""
+    Synchronizes local and remote RPC processes.
+
+    This will block until all local and remote RPC processes specified under worker_names
+    reach this method to wait for all outstanding work to complete.
+
+    Args:
+        worker_names (List[str]): The set of workers to synchronize.
+
+    """
+    try:
+        _all_gather(None, set(worker_names), timeout=DEFAULT_SHUTDOWN_TIMEOUT)
+    except RuntimeError as ex:
+        logger.error(
+            f"Failed to respond to 'Shutdown Proceed' in time, got error {ex}"
```

Comment:
If this is being used for non-shutdown case, this error might be a bit confusing (says "shutdown proceed")?

Reply:
Good catch, I think i had copy-pasted from shutdown() 🙂, will change it.